### PR TITLE
airbyte-ci: ignore archived connectors

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes https://github.com/airbytehq/airbyte-internal-issues/issues/8942

This change means we won't auto select any archived connectors to run `connectors` subcommand like publish and test.
Archived connectors could still be selected with explicit `--name` option, but they won't be selected according to file modification when using `--modified`.
This is to avoid publishing (and testing) archived connectors.

### CODE REVIEW

**Comment by alafanechere** [comment]
* **#43426** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/43426?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alafanechere and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by natikgadzhi — APPROVED** [review]
Curious: it looks like this change only excludes archived connectors form `--modified` selection? Is that the intent? (unblocking in case it is).

I assume `airbyte-ci connectors --support-level archived` would work to select them as well?

File: airbyte-ci/connectors/pipelines/pipelines/helpers/connectors/modifed.py [hunk 1]
```diff
@@ -27,6 +27,9 @@ def _find_modified_connectors(
     modified_connectors = set()
 
     for connector in all_connectors:
+        if connector.support_level == "archived":
+            main_logger.info(f"Skipping connector '{connector}' due to 'archived' support level.")
```
Comment (natikgadzhi) [inline airbyte-ci/connectors/pipelines/pipelines/helpers/connectors/modifed.py:31]: Ask: let's add that caveat to the help message of the `--modified` flag.

**Review by natikgadzhi — APPROVED** [review]
Curious: it looks like this change only excludes archived connectors form `--modified` selection? Is that the intent? (unblocking in case it is).

I assume `airbyte-ci connectors --support-level archived` would work to select them as well? 

Left one ask, but overall this seems like a good step. Let's add the help message clarification!

**Comment by alafanechere** [comment]
> Curious: it looks like this change only excludes archived connectors form `--modified` selection? Is that the intent? (unblocking in case it is).
> 
> 
> 
> I assume `airbyte-ci connectors --support-level archived` would work to select them as well? 

Yes the indent is to only deselect them on automatic selection with --modified. Others explicit (and likely local) execution with --name or --support-level will still allow selecting archived connectors

**Pull request merged**